src/main.ts

```diff
@@ -64,19 +72,31 @@ async function run() {
     try {
         const token = core.getInput('repo_token', { required: true });
         const file = core.getInput('file', { required: true });
-        const asset_name = core.getInput('asset_name', { required: true });
+        const file_glob = core.getInput('file_glob');
         const tag = core.getInput('tag', { required: true }).replace("refs/tags/", "");
         const overwrite = core.getInput('overwrite');
 
-        if (!fs.existsSync(file)) {
-            core.setFailed(`File ${file} wasn't found.`);
-        }
-
         const octokit = new github.GitHub(token);
         const context = github.context;
-
         const release = await get_release_by_tag(tag, octokit, context);
-        await upload_to_release(release, file, asset_name, tag, overwrite, octokit, context);
+
+        if (file_glob === "true") {
+            const files = glob.sync(file);
+            if (files.length > 0) {
```

Comment:
Or actually... one last thing! Any chance you can make this into a non-indexed loop and use an iterator rather? I think it would make this a bit more readable.

Reply:
sorry got busy last week, i've updated the pr with the changes. let me know if that works! :